models/nexusmutual/ethereum/capital_pool/nexusmutual_ethereum_capital_pool_totals.sql

```diff
@@ -0,0 +1,500 @@
+    coalesce(drt.reth_total * p_avg_reth.price_usd, 0) as avg_reth_usd_total,
+    coalesce(drt.reth_total * p_avg_reth.price_usd / p_avg_eth.price_usd, 0) as avg_reth_eth_total,
+    -- USDC
+    coalesce(drt.usdc_total, 0) as usdc_total,
+    coalesce(drt.usdc_total * p_avg_usdc.price_usd, 0) as avg_usdc_usd_total,
+    coalesce(drt.usdc_total * p_avg_usdc.price_usd / p_avg_eth.price_usd, 0) as avg_usdc_eth_total,
+    -- Cover Re USDC investment
+    coalesce(drt.cover_re_usdc_total, 0) as cover_re_usdc_total,
+    coalesce(drt.cover_re_usdc_total * p_avg_usdc.price_usd, 0) as avg_cover_re_usdc_usd_total,
+    coalesce(drt.cover_re_usdc_total * p_avg_usdc.price_usd / p_avg_eth.price_usd, 0) as avg_cover_re_usdc_eth_total,
+    -- AAVE positions
+    coalesce(drt.aave_collateral_weth_total, 0) as aave_collateral_weth_total,
+    coalesce(drt.aave_collateral_weth_total * p_avg_eth.price_usd, 0) as avg_aave_collateral_weth_usd_total,
+    coalesce(drt.aave_debt_usdc_total, 0) as aave_debt_usdc_total,
+    coalesce(drt.aave_debt_usdc_total * p_avg_usdc.price_usd, 0) as avg_aave_debt_usdc_usd_total,
+    coalesce(drt.aave_debt_usdc_total * p_avg_usdc.price_usd / p_avg_eth.price_usd, 0) as avg_aave_debt_usdc_eth_total
+  from daily_running_totals drt
+    inner join daily_avg_eth_prices p_avg_eth on drt.block_date = p_avg_eth.block_date
+    left join daily_avg_dai_prices p_avg_dai on drt.block_date = p_avg_dai.block_date
+    left join daily_avg_reth_prices p_avg_reth on drt.block_date = p_avg_reth.block_date
+    left join daily_avg_usdc_prices p_avg_usdc on drt.block_date = p_avg_usdc.block_date
+)
+
+select
+  block_date,
+  avg_eth_usd_price,
+  -- Capital Pool totals
+  eth_total + nxmty_eth_total + steth_total + avg_dai_eth_total + avg_reth_eth_total + avg_usdc_eth_total
+    + avg_cover_re_usdc_eth_total + aave_collateral_weth_total + avg_aave_debt_usdc_eth_total as avg_capital_pool_eth_total,
+  avg_eth_usd_total + avg_nxmty_usd_total + avg_steth_usd_total + avg_dai_usd_total + avg_reth_usd_total + avg_usdc_usd_total
+    + avg_cover_re_usdc_usd_total + avg_aave_collateral_weth_usd_total + avg_aave_debt_usdc_usd_total as avg_capital_pool_usd_total,
+  -- ETH
+  eth_total,
+  avg_eth_usd_total,
+  -- DAI
+  dai_total,
+  avg_dai_usd_total,
+  avg_dai_eth_total,
+  -- NXMTY
+  nxmty_total,
+  nxmty_eth_total,
+  avg_nxmty_usd_total,
+  -- stETH
+  steth_total,
+  avg_steth_usd_total,
+  -- rETH
+  reth_total,
+  avg_reth_usd_total,
+  avg_reth_eth_total,
+  -- USDC
+  usdc_total,
+  avg_usdc_usd_total,
+  avg_usdc_eth_total,
+  -- Cover Re USDC investment
+  cover_re_usdc_total,
+  avg_cover_re_usdc_usd_total,
+  avg_cover_re_usdc_eth_total,
+  -- AAVE positions
+  aave_collateral_weth_total,
+  avg_aave_collateral_weth_usd_total,
+  aave_debt_usdc_total,
+  avg_aave_debt_usdc_usd_total,
+  avg_aave_debt_usdc_eth_total
+from daily_running_totals_enriched
+{% if is_incremental() %}
+where {{ incremental_predicate('block_date') }}
+{% endif %}
```

Comment:
are you not able to filter incrementally in CTEs above, since you need to sum() for all history? is that why it happens only at the end here?

Reply:
Unfortunately not easily with all these running totals - many of them depending on each other. I know it seems like a waste to keep re-running the whole thing over and over again, but if I find a better way later on, will re-visit it 🤞

Reply:
all good, just confirming. i also need to take the time to find a good solution for that, as it's come up a few times.